### PR TITLE
gpt4all: rename `gpt4all-chat` in `gpt4all`

### PR DESCRIPTION
**Pull request by drupol**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by philiptaron** [comment]
Do you need to provide an alias (see: https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/aliases.nix)?

I think you do...

**Review by philiptaron — CHANGES_REQUESTED** [review]
Please add a renamed alias in https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/aliases.nix per that file's conventions.

Other changes look good.

**Comment by drupol** [comment]
Job done, thanks for the hint!

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -411,6 +411,9 @@ mapAliases ({
   govendor = throw "'govendor' has been removed as it is no longer maintained upstream, please use Go modules instead"; # Added 2023-12-26
   gometer = throw "gometer has been removed from nixpkgs because goLance stopped offering Linux support"; # Added 2023-02-10
   gprbuild-boot = gnatPackages.gprbuild-boot; # Added 2024-02-25;
+
+  gpt4all-chat = throw "gpt4all-chat has been renamed to gpt4all"; # Added 2024-02-27
```
Comment (philiptaron) [inline pkgs/top-level/aliases.nix:415]: I think you just alias first, especially in a project that doesn't have on-disk state. See https://github.com/NixOS/nixpkgs/commit/0c634976411118a2aace4fb6fe70cb398b73bc9a which gave about three years of aliasing prior to throwing.

**Review by philiptaron — CHANGES_REQUESTED** [review]
I could be wrong, but I think an alias is in order vs. a `throw`.

File: pkgs/top-level/aliases.nix (hunk 1 above)
Comment (philiptaron) [reply pkgs/top-level/aliases.nix:415]: Hmmm. I thought it was older than that, but I'm just wrong. Let's roll with this version.

**Review by philiptaron — APPROVED** [review]

**Pull request merged**